src/Common/Provider/AbstractHttpProvider.php

```diff
@@ -35,13 +36,11 @@
     private $messageFactory;
 
     /**
-     * @param HttpClient          $client
-     * @param MessageFactory|null $factory
+     * @param HttpClient $client
      */
-    public function __construct(HttpClient $client, MessageFactory $factory = null)
+    public function __construct(HttpClient $client)
```

Comment:
Please make this change in a separate issue/PR. It requires far more discussion than your "add locale feature"